### PR TITLE
Host change events now parse the value included in the event

### PR DESCRIPTION
**Pull request by thrawn01**
## Purpose
* Vulcand can report errors if a PUT is quickly followed by a DEL in Etcd. This change uses the value in the PUT event to parse the new host entry to avoid this scenario and avoid an additional call to Etcd.
* Now indicates the offending etcd key when reporting JSON parse errors.

### CODE REVIEW

File: engine/etcdng/v3/etcd.go [hunk 1]
```diff
@@ -281,14 +281,17 @@ func (n *ng) GetHost(key engine.HostKey) (*engine.Host, error) {
...
 	}
 
+	return n.parseHost(host, key.Name)
+}
+
+func (n *ng) parseHost(host *host, name string) (*engine.Host, error) {
 	var keyPair *engine.KeyPair
 	if len(host.Settings.KeyPair) != 0 {
 		if err := n.openSealedJSONVal(host.Settings.KeyPair, &keyPair); err != nil {
 			return nil, err
```
Comment (horkhe) [inline engine/etcdng/v3/etcd.go]: Could you please also wrap this error.

File: engine/etcdng/v3/etcd.go [hunk 2]
```diff
@@ -673,7 +676,13 @@ func (n *ng) parseHostChange(e *etcd.Event) (interface{}, error) {
 
 	switch e.Type {
 	case etcd.EventTypePut:
-		host, err := n.GetHost(engine.HostKey{Name: hostname})
+		var jsonHost host
+		err := json.Unmarshal(e.Kv.Value, &jsonHost)
+		if err != nil {
+			return nil, err
```
Comment (horkhe) [inline engine/etcdng/v3/etcd.go]: ...and this

File: engine/etcdng/v3/etcd.go [hunk 3]
```diff
@@ -673,7 +676,13 @@ func (n *ng) parseHostChange(e *etcd.Event) (interface{}, error) {
...
+		}
+
+		host, err := n.parseHost(&jsonHost, hostname)
 		if err != nil {
 			return nil, err
```
Comment (horkhe) [inline engine/etcdng/v3/etcd.go]: ...and this

**Comment by thrawn01** [comment]
This is different, the changes were made, but the comments are still showing the old diffs.

**Pull request merged**